src/routing_table/prefix.rs

```diff
@@ -94,9 +94,25 @@ impl<T: Clone + Copy + Default + Binary + Xorable> Prefix<T> {
     }
 
     /// Compares the distance of `self` and `other` to `target`. Returns `Less` if `self` is closer,
-    /// `Greater` if `other` is closer, and `Equal` if `self.name == other.name`.
+    /// `Greater` if `other` is closer, and compares the prefix directly if of equal distance
+    /// (this is to make sorting deterministic).
     pub fn cmp_distance(&self, other: &Self, target: &T) -> Ordering {
-        target.cmp_distance(&self.name, &other.name)
+        if self.is_compatible(&other) {
+            Ord::cmp(&self.name, &other.name)
```

Comment:
Or `bit_count` instead of `name`? Then it wouldn't depend on the canonical name and different prefixes would be guaranteed to not return `Equal` here: Currently `000` and `00` are still considered equal.